src/test/mod.rs

```diff
@@ -220,3 +186,410 @@ fn structs() {
+            );
+
+            assert_eq!(
+                *run(r#" "hi".shell("echo", "bash -c")"#),
+                FenderValue::make_string("hi\n".into())
+            );
+        }
+    }
+
+    mod val_operations {
+        use super::*;
+
+        #[test]
+        fn len() {
+            assert_eq!(*run(r#""hello".len()"#), FenderValue::Int(5));
+            assert_eq!(*run(r#"[1, 2, 3, 4, 5].len()"#), FenderValue::Int(5));
+
+            assert!(matches!(*run(r#"2.len()"#), FenderValue::Error(_)))
+        }
+
+        #[test]
+        fn swap() {
+            assert_eq!(
+                *run(r#"[0,1,2,3]"#),
+                FenderValue::make_list(
+                    (0..=3)
+                        .into_iter()
+                        .map(|i| FenderValue::Int(i).into())
+                        .collect()
+                )
+            );
+
+            assert_eq!(
+                *run(r#"[0,1,2,3].swap(0, 3)"#),
+                FenderValue::make_list(
+                    ([3, 1, 2, 0])
+                        .into_iter()
+                        .map(|i| FenderValue::Int(i).into())
+                        .collect()
+                )
+            );
+            match *run(r#"[0,1,2,3].swap(0, "3")"#) {
+                FenderValue::Error(_) => (),
+                _ => panic!(),
+            }
+        }
+
+        #[test]
+        #[should_panic]
+        fn swap_panic() {
+            run(r#"[0,1,2,3].swap(0, 9)"#);
+        }
+
+        #[test]
+        fn shuffle() {
+            let input_list = (0..100).collect::<Vec<_>>();
+            let test_prog = format!(
+                "$ordered = {:?}; $new = ordered.shuffle(); return [ordered, new]",
+                input_list
+            );
+
+            let results = match (*run(&test_prog)).clone() {
+                FenderValue::List(l) => ((*l[0]).to_string(), (*l[1]).to_string()),
+                _ => unreachable!(),
+            };
+            assert_ne!(results.0, format!("{:?}", input_list));
+            assert_ne!(results.1, format!("{:?}", input_list));
+            assert_eq!(results.0, results.1);
+
+            // -_- I hate this
+        }
+
+        #[test]
+        fn shuffled() {
+            let input_list = (0..100).collect::<Vec<_>>();
+            let test_prog = format!(
+                "$ordered = {:?}; $new = ordered.getShuffled(); return [ordered, new]",
+                input_list
+            );
+
+            let results = match (*run(&test_prog)).clone() {
+                FenderValue::List(l) => ((*l[0]).to_string(), (*l[1]).to_string()),
+                _ => unreachable!(),
+            };
+            assert_eq!(results.0, format!("{:?}", input_list));
+            assert_ne!(results.1, format!("{:?}", input_list));
+            assert_ne!(results.0, results.1);
+        }
+
+        #[test]
+        fn rand() {
+            assert!(matches!(*run(r#"rand()"#), FenderValue::Float(_)))
```

Comment:
Maybe run it twice and assert that the returned values are not equal?

Reply:
That could randomly be false

Reply:
The odds would be so low that it would essentially never happen - if you really want to make it even more unlikely, you could generate three random numbers and assert that they're not all equal to each other